app/routes/rental_routes.py

```diff
@@ -0,0 +1,79 @@
+from app import db
+from flask import Blueprint, jsonify, request
+from app.models.rental import Rental
+from app.models.customer import Customer
+from app.models.video import Video
+from datetime import timedelta, datetime
+from app.models.helper import invalid_rental_data
+
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+@rentals_bp.route("/check-out", methods=["POST"])
+def post_rental():
+    request_body = request.get_json()
+    invalid_request = invalid_rental_data(request_body)
+    if invalid_request:
+        return jsonify(invalid_request), 400
+
+    one_video = Video.query.get(request_body["video_id"])
+    one_customer = Customer.query.get(request_body["customer_id"])
+    if one_video is None:
+        return jsonify({"Bad Request": f"Video not found."}), 404
+    elif one_customer is None:
+        return jsonify({"Bad Request": f"Customer not found."}), 404
+    video_current_checked_out = Rental.query.filter_by(checked_out=True, video_id=one_video.id).count()
+    available_inventory = one_video.total_inventory - video_current_checked_out
+    if available_inventory == 0:
+        return jsonify({"message": "Could not perform checkout"}), 400
+    new_rental = Rental(
+        customer_id=one_customer.id,
+        video_id = one_video.id,
+        due_date = (datetime.now(tz=None) + timedelta(days=7)),
+        checked_out = True
+    )
+    db.session.add(new_rental)
+    db.session.commit()
+    cust_current_checked_out = Rental.query.filter_by(customer_id=one_customer.id, checked_out=True).count()
+    available_inventory -=1
+    response_body = {
+        "customer_id": one_customer.id,
+            "video_id": one_video.id,
+            "due_date": new_rental.due_date,
+            "videos_checked_out_count": cust_current_checked_out,
+            "available_inventory": available_inventory
+    }
+    return jsonify(response_body), 200
+
+@rentals_bp.route("/check-in", methods=["POST"])
+def post_rental_return():
+    request_body = request.get_json()
+    invalid_request = invalid_rental_data(request_body)
+    if invalid_request:
+        return jsonify(invalid_request), 400
+    
+    one_video = Video.query.get(request_body["video_id"])
+    one_customer = Customer.query.get(request_body["customer_id"])
+    if one_video is None:
+        return jsonify({"Bad Request": f"Video not found."}), 404
+    elif one_customer is None:
+        return jsonify({"Bad Request": f"Customer not found."}), 404
+    video_current_checked_in = Rental.query.filter_by(checked_out=True).count()
+    video_checked_out = Rental.query.filter_by(checked_out=False).count()
```

Comment:
Here, we should include the video id in the query for getting the count of checked out rentals for the video and those that _are_ checked out. We should also look this up _after_ we have checked in the video so the counts line up. I think the tests work for this since they expect to check in a video (so the checked out count should be 0 and everything should be available). Since this code is looking for the count of checked in rentals before we checked any in, it still ends up getting a count of 0, but it's getting the correct result (in this situation only) for an erroneous reason.